base/task.jl

```diff
@@ -365,6 +365,67 @@ function wait(t::Task)
     nothing
 end
 
+# Wait multiple tasks
+waitany(tasks) = _wait_multiple(tasks)
+waitall(tasks; failfast=false) = _wait_multiple(tasks; all=true, failfast=failfast)
+
+function _wait_multiple(waiting_tasks; all=false, failfast=false)
+    foreach(waiting_tasks) do @nospecialize t
+        if !(t isa Task)
+            error("Expected an iterator of `Task` object")
+        end
+    end
+
+    chan = Channel{Tuple{Int,Task}}(Inf)
+    tasks = Task[]
+    done_mask = Bool[]
+    exception = false
+    nremaining::Int = 0
+
+    for (i, t) in enumerate(waiting_tasks)
+        t = t::Task
+        push!(tasks, t)
+        if istaskdone(t)
+            push!(done_mask, true)
+            exception |= istaskfailed(t)
+        else
+            nremaining += 1
+            push!(done_mask, false)
+            schedule(Task(() -> begin
+                _wait(t)
+                put!(chan, (i, t))
+            end))
```

Comment:
performance nit: this is a more optimized internal version of the same:
```suggestion
            t2 = @task put!(chan, (i, t))
            t2.sticky = false
            _wait2(t, t2)
```

Reply:
The commit 1bd4973 includes this suggestion.